app/upload/directives/audio-file.component.ts

```diff
@@ -0,0 +1,100 @@
+import {Component, Input, Output, EventEmitter} from 'angular2/core';
+import {FileSizePipe} from '../../shared/file/filesize.pipe';
+import {AudioModel} from '../models/audio.model';
+
+@Component({
+  host: {
+    '[class.canceled]': 'canceled'
+  },
+  pipes: [FileSizePipe],
+  selector: 'audio-file',
+  styleUrls: ['app/upload/directives/audio-file.component.css'],
+  template: `
+    <div class="reorder">
+      <i *ngIf="!canceled && !audio.isUploading" class="icon-menu drag-handle"></i>
+    </div>
+
+    <div class="main">
+
+      <div class="type">
+        <span>Segment</span>
+        <i class="icon-down-dir"></i>
+      </div>
+
+      <div class="info">
+        <span>{{audio.filename}}</span>
+        <span class="size">({{audio.size | filesize}})</span>
+      </div>
+
+      <div *ngIf="canceled" class="progress canceled">
+        <p *ngIf="audio.isUploading">Upload Canceled</p>
+        <p *ngIf="!audio.isUploading">File Deleted</p>
+      </div>
+
+      <div *ngIf="!canceled && !audio.isDone" [class]="progressClass">
+        <p>{{statusString}}</p>
+        <div *ngIf="audio.isError" class="retry">
+          <a class="icon-cw" href="#" (click)="onRetry($event)">Try Again</a>
+        </div>
+        <div *ngIf="!audio.isError && audio.isUploading" class="meter">
+          <span [style.width.%]="audio.progress * 100"></span>
```

Comment:
For some reason, phantom was choking on the percentage pipe that was in there before.  But everything is happy with the `[style.width.%]` syntax.